unit-tests/src/test/java/org/eclipse/collections/impl/list/immutable/AbstractImmutableListTestCase.java

```diff
@@ -305,14 +305,28 @@ public void distinct()
     @Test
     public void distinctWithHashingStrategy()
     {
-        FastList<String> strings = FastList.newListWith("A", "b", "a", "c", "B", "D", "e", "D", "e", "E");
+        MutableList<String> strings = Lists.mutable.with("A", "b", "a", "c", "B", "D", "e", "D", "e", "E");
         ImmutableList<Integer> integers = this.classUnderTest();
         ImmutableList<String> letters =
                 strings.subList(0, integers.size()).toImmutable().distinct(HashingStrategies.fromFunction(String::toLowerCase));
         List<String> expectedLetters = strings.subList(0, integers.size()).distinct(HashingStrategies.fromFunction(String::toLowerCase));
         Assert.assertEquals(expectedLetters, letters);
     }
 
+    /**
+     * @since 9.0.
+     */
+    @Test
+    public void distinctBy()
+    {
+        MutableList<String> strings = Lists.mutable.with("A", "b", "a", "c", "B", "D", "e", "D", "e", "E");
+        ImmutableList<Integer> integers = this.classUnderTest();
+        ImmutableList<String> letters =
+                strings.subList(0, integers.size()).toImmutable().distinctBy(String::toLowerCase);
+        List<String> expectedLetters = strings.subList(0, integers.size()).distinctBy(String::toLowerCase);
```

Comment:
This test looks weird to me. You are testing `distinctBy()` with `distinctBy()`. 

`List<String> expected = strings.collect(String::toLowerCase).distinct();`

`List<String> actual = letters.collect(String::toLowerCase);`

It is sort of difficult to test with capital and lower case.

May be have the expected pre-populated like you do in a test below.